readthedocs/vcs_support/backends/git.py

```diff
@@ -523,8 +537,84 @@ def commit(self):
         return None
 
     @property
-    def submodules(self):
-        return list(self._repo.submodules)
+    def submodules(self) -> Iterable[GitSubmodule]:
+        r"""
+        Return a iterable of submodules in this repository.
+
+        In order to get the submodules URLs and paths without initializing them,
+        we parse the .gitmodules file. For this we make use of the
+        ``git config --get-regexp`` command.
+
+        Keys and values from the config can contain spaces,
+        in order to parse the output unambiguously, we use the
+        ``--null`` option to separate each result with a null character,
+        and each key and value with a newline character.
+
+        The command will produce an output like this:
+
+        .. code-block:: text
+
+           submodule.submodule-1.url\nhttps://github.com/\0
+           submodule.submodule-1.path\nsubmodule-1\0
+           submodule.submodule-2.path\nsubmodule-2\0
+           submodule.submodule-2.url\nhttps://github.com/\0
+           submodule.submodule-3.url\nhttps://github.com/\0
+           submodule.submodule-4.path\n\0
+
+        .. note::
+
+           - In the example each result is put in a new line for readability.
+           - Isn't guaranteed that the url and path keys will appear next to each other.
+           - Isn't guaranteed that that all submodules will have a url and path.
+
+        """
+        exit_code, stdout, _ = self.run(
+            "git",
+            "config",
+            "--null",
+            "--file",
+            ".gitmodules",
+            "--get-regexp",
+            # Get only the URL and path keys of each submodule.
+            r"^submodule\..*\.(url|path)$",
+            record=False,
+        )
+        if exit_code != 0:
+            return []
```

Comment:
It would be good to `log.warning` here as well. We don't fail the build, but we can get some log lines to debug this more in case there is an issue in our side.

Reply:
The command will fail if the project doesn't have submodules (the .gitmodules file doesn't exist).

Reply:
Gotcha!

Reply:
```suggestion
        if exit_code != 0:
            # The command fails if the project doesn't have submodules (the .gitmodules file doesn't exist).
            return []
```